consensus/poet/core/sawtooth_poet/poet_consensus/utils.py

```diff
@@ -103,3 +105,19 @@ def build_certificate_list(block_header,
         LOGGER.error('Error getting block: %s', ke)
 
     return list(certificates)
+
+
+def state_view_for_block(block, state_view_factory):
+    """Returns a read-only state view object for a particular block
+
+    Args:
+        block (BlockWrapper): The block for which to return the state view.
+        state_view_factory (StateViewFactor): The state view factory
+            to be used to create the state view object
+
+    Returns:
+        StateView: The read-only state view for the block
+    """
+    state_root_hash = \
+        block.state_root_hash if block is not None else INIT_ROOT_KEY
```

Comment:
I don't see any changes to `StateViewFactory`.  Do you not need to worry about this anymore?